launch_frontend/launch_frontend/entity.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for Entity class."""
+
+import io
+from typing import Any
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Union
+
+from launch.utilities import is_a
+
+from pkg_resources import iter_entry_points
+
+frontend_entities = None
+
+
+class Entity:
+    """Single item in the intermediate front_end representation."""
+
+    @staticmethod
+    def load(
+        file: Union[str, io.TextIOBase],
+        parent: 'Entity' = None
+    ) -> 'Entity':
+        """Return an entity loaded with a markup file."""
+        # frontend_entities is not global to avoid a recursive import
+        global frontend_entities
+        if frontend_entities is None:
+            frontend_entities = {
+                entry_point.name: entry_point.load()
+                for entry_point in iter_entry_points('launch_frontend.entity')
+            }
+        if is_a(file, str):
+            # This automatically recognizes 'file.xml' or 'file.launch.xml'
+            # as a launch file using the xml frontend.
+            frontend_name = file.rsplit('.', 1)[1]
+            if frontend_name in frontend_entities:
+                return frontend_entities[frontend_name].load(file)
```

Comment:
@ivanpauno this is something that could be done by a method in a `Parser` subclass that's specific to each frontend instead of making assumptions here.

Reply:
If you have an `import` action (`ImportLaunchDescription`) as part of one launch frontend file (eg: a xml launch file), the file imported could been written in any of the other supported front-ends.

So the abstract-parser should infer what kind of file is being imported, and then delegate the actual parsing to the correct parser implementation. As I think loading could be considered part of the work that `Entity` does, I added it there.

Two ways of inferring what front-end file is being loaded are possible:
- Having strict launch extension filenaming rules.
- Do a force-brute load with all the existing front-ends, if one of them not fails use it. This would end with non-clear error messages in the case the file is badly formed.

Now, I'm doing both things: first check extension, then brute-force load.